src/components/Modal/ModalSelectAssets.tsx

```diff
@@ -35,51 +41,83 @@ export const ModalSelectAssets = () => {
 
   const handleSearchClear = () => setSearchValue("")
 
-  const filterPattern = (asset: BaseTokenInfo) =>
-    asset.name
+  const filterPattern = (asset: SelectItemToken) =>
+    asset.token.name
       .toLocaleUpperCase()
       .includes(deferredQuery.toLocaleUpperCase()) ||
-    asset.chainName
-      .toLocaleUpperCase()
-      .includes(deferredQuery.toLocaleUpperCase())
+    (isBaseToken(asset.token)
+      ? asset.token.chainName
+          .toLocaleUpperCase()
+          .includes(deferredQuery.toLocaleUpperCase())
+      : true)
+
+  const handleSelectToken = (token: SelectItemToken) => {
+    if (modalType !== ModalType.MODAL_SELECT_ASSETS) {
+      throw new Error("Invalid modal type")
+    }
 
-  const handleSelectToken = (token: BaseTokenInfo) => {
-    onCloseModal({
-      ...(payload as { fieldName: string }),
-      modalType,
-      token,
-    })
+    const newPayload: ModalSelectAssetsPayload = {
+      ...(payload as ModalSelectAssetsPayload),
+      modalType: ModalType.MODAL_SELECT_ASSETS,
+      token: token.token,
```

Comment:
The function property `token` seems to have some issues in the token interface. I would suggest renaming it to something like`token: selectedItem.token`.